### PR TITLE
Print function for DerivativeForm

### PR DESCRIPTION
**Pull request by bergbauer**
Used for the output in https://github.com/dealii/dealii/pull/16081#issuecomment-1743684564

### CODE REVIEW

File: include/deal.II/base/derivative_form.h [hunk 1]
```diff
@@ -428,6 +428,30 @@ DerivativeForm<order, dim, spacedim, Number>::memory_consumption()
...
+ * Output operator for DerivativeForm. Print the elements consecutively, with a
+ * space in between, two spaces between rank 1 subtensors, three between rank 2
+ * and so on.
+ *
+ * @relatesalso DerivativeForm
+ */
+template <int order, int dim, int spacedim, typename Number>
+inline std::ostream &
+operator<<(std::ostream                                       &out,
+           const DerivativeForm<order, dim, spacedim, Number> &df)
```
Comment (peterrum) [inline include/deal.II/base/derivative_form.h:441]: Is there somewhere a similar function that you have adopted here? Could you point me out to that one?

**Review by peterrum — APPROVED** [review]
Thanks! Made the review easier :)

File: include/deal.II/base/derivative_form.h [hunk 2]
```diff
@@ -428,6 +428,30 @@ DerivativeForm<order, dim, spacedim, Number>::memory_consumption()
...
+{
+  for (unsigned int i = 0; i < spacedim; ++i)
+    {
+      out << df[i];
```
Comment (bergbauer) [inline include/deal.II/base/derivative_form.h:445]: This line uses the referenced `Tensor` functions. @peterrum

**Review by bangerth — APPROVED** [review]

**Comment by bangerth** [comment]
Usually, we would ask for a test, but I understand that that function will be used in a different patch?

Oh, separately, I clicked "merge" too quickly. Can you, in a separate PR, add an entry to doc/news/changes/minor?

**Comment by bergbauer** [comment]
> Usually, we would ask for a test, but I understand that that function will be used in a different patch?

I have only used this in #16081 for debugging purposes (see https://github.com/dealii/dealii/pull/16081#issuecomment-1743684564). The follow-up #16089 also adds a test.

**Pull request merged**